### PR TITLE
Adding package with lib reSIProcate, version 1.12.0

### PR DESCRIPTION
**Pull request by AlexeiKislinskii10**
Specify library name and version:  **reSIProcate/1.12.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/resiprocate/all/conanfile.py
Comment (prince-chrismc) [inline recipes/resiprocate/all/conanfile.py:14]: https://github.com/resiprocate/resiprocate/blob/76508df8e06a4eb001e499bb2557853549f397e0/COPYING#L16
https://spdx.org/licenses/VSL-1.0.html

```suggestion
    license = "VSL-1.0"
```

File: recipes/resiprocate/all/test_package/test_package.cpp
Comment (prince-chrismc) [inline recipes/resiprocate/all/test_package/test_package.cpp:27]: would it be easier to embed the file into the code and have a minimal test package?

File: recipes/resiprocate/all/conanfile.py
Comment (prince-chrismc) [inline recipes/resiprocate/all/conanfile.py:38]: You _should_ use autotool to install https://github.com/resiprocate/resiprocate/blob/76508df8e06a4eb001e499bb2557853549f397e0/INSTALL#L6

Example:
https://github.com/conan-io/conan-center-index/blob/b804571d270b799eeaf3b7b83634ae1f7268083d/recipes/libuuid/all/conanfile.py#L54
https://github.com/conan-io/conan-center-index/blob/b804571d270b799eeaf3b7b83634ae1f7268083d/recipes/libuuid/all/conanfile.py#L33

File: recipes/resiprocate/all/conanfile.py
Comment (AlexeiKislinskii10) [reply recipes/resiprocate/all/conanfile.py]: Thanks!

File: recipes/resiprocate/all/test_package/test_package.cpp
Comment (AlexeiKislinskii10) [reply recipes/resiprocate/all/test_package/test_package.cpp]: It make sense, fixed.

File: recipes/resiprocate/all/test_package/test_package.cpp
Comment (prince-chrismc) [inline recipes/resiprocate/all/test_package/test_package.cpp:10]: In case you did know about it, https://en.cppreference.com/w/cpp/language/string_literal number 6 is a "raw string literal" might have made this a little easier

File: recipes/resiprocate/all/conanfile.py
Comment (prince-chrismc) [inline recipes/resiprocate/all/conanfile.py:18]: This is the usually pattern we have, but it will require you to update other parts too

```suggestion
    @property
    def _source_subfolder(self):
        return "source_subfolder"

    def source(self):
        tools.get(**self.conan_data["sources"][self.version])
        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
```

File: recipes/resiprocate/all/test_package/test_package.cpp
Comment (AlexeiKislinskii10) [reply recipes/resiprocate/all/test_package/test_package.cpp:10]: Yep, I know about raw string letteral, but as for me it would me much easier to predict line ending style in case of line concatenation.

Comment (prince-chrismc) [reply recipes/resiprocate/all/test_package/test_package.cpp:10]: I just noticed the SDP content, that needs to be `\r\n` and the C++ ISO says raw strings are `\n` only  so sport on ! xD

**Comment by uilianries** [comment]
@AlexeiKislinskii10 , please consider https://github.com/AlexeiKislinskii10/conan-center-index/pull/2

**Comment by AlexeiKislinskii10** [comment]
> @AlexeiKislinskii10 , please consider [AlexeiKislinskii10#2](https://github.com/AlexeiKislinskii10/conan-center-index/pull/2)

Glad to see that Your are active! I'm started work basing on Your [conanfile.py](https://github.com/uilianries/conan-resiprocate/blob/release/1.10.2/conanfile.py)! Just couldn't test all this options myself and remove them=(

File: recipes/resiprocate/all/conanfile.py
Comment (AlexeiKislinskii10) [reply recipes/resiprocate/all/conanfile.py]: Thanks to @uilianries it's fixed!

File: recipes/resiprocate/all/conanfile.py
Comment (AlexeiKislinskii10) [reply recipes/resiprocate/all/conanfile.py:42]: Thanks to @uilianries it's fixed!

**Comment by uilianries** [comment]
@AlexeiKislinskii10 I saw reSIProcate offers .sln (Visual Studio solution) files, and Conan MSBuild() Conan helper should be able to execute, but I didn't check. Maybe, it will be easier using MSBuild instead of using MSYS

**Comment by uilianries** [comment]
@AlexeiKislinskii10 I saw reSIProcate offers .sln (Visual Studio solution) files, and Conan MSBuild() Conan helper should be able to execute, but I didn't check. Maybe, it will be easier using MSBuild instead of using MSYS

**Comment by AlexeiKislinskii10** [comment]
> @AlexeiKislinskii10 I saw reSIProcate offers .sln (Visual Studio solution) files, and Conan MSBuild() Conan helper should be able to execute, but I didn't check. Maybe, it will be easier using MSBuild instead of using MSYS

MSBuild failed to run sln from this project, it generate exception: "Visual Studio 15 Cannot recognize the Windows subsystem". Also test with clear run of msbuild.exe showed that we need to detect user toolset and win SDK before build, in other cases msbuild even with auto upgrade parameter didn't build resiprocate solution

**Comment by prince-chrismc** [comment]
you need to fix the commit history, https://github.com/conan-io/conan-center-index/pull/3432#issuecomment-722676951

**Comment by AlexeiKislinskii10** [comment]
For now resiprocate is ready for use on Linux(tested on CentOS 8 and Debian 9). Windows is disabled because of errors in MSBuild tool, and on Mac because of errors in process of test_package running. For now I haven't access to mac so I don't know how to fix package for it, but with Windows I've got plan and will get back to it soon(hope about 3 - 6 weeks). Also maintainer of resiprocate announced resiprocate 1.13.0, so got plans about working on this package too. Hope package only for Linux whould be accepted for conan center, because I really need it officially. Sorry about too height load of build-bot, now I've got experience with it an next releases would be with more clear commit history!=)

File: recipes/resiprocate/all/conanfile.py
Comment (prince-chrismc) [inline recipes/resiprocate/all/conanfile.py:42]: you can pass the version at the command-line `conan create . 1.29.0@`

```suggestion
```

File: recipes/resiprocate/all/conanfile.py
Comment (prince-chrismc) [inline recipes/resiprocate/all/conanfile.py:44]: I dont think this needed in CCI

```suggestion
```

Comment (AlexeiKislinskii10) [reply recipes/resiprocate/all/conanfile.py:44]: It just make easier command `conan source .`

File: recipes/resiprocate/all/conanfile.py
Comment (prince-chrismc) [inline recipes/resiprocate/all/conanfile.py:6]: https://cpp.libhunt.com/conan-changelog/1.29.1

https://github.com/conan-io/conan-center-index/pull/3432/files#diff-fadd84d6f3b955981505cfa3e8a382f561f228388350c42cdd88a3d63b3e100fR73

```suggestion


required_conan_version = ">=1.29.1"

class ResiprocateConan(ConanFile):
```

File: recipes/resiprocate/all/conanfile.py
Comment (prince-chrismc) [inline recipes/resiprocate/all/conanfile.py:2]: Unused I believe (apply if there's any problems)

```suggestion
```

**Review by prince-chrismc — APPROVED** [review]
:rocket: save the rest for the next PR!

LGTM

**Review by prince-chrismc — DISMISSED** [review]

**Comment by AlexeiKislinskii10** [comment]
@uilianries can You look at this PR again, I think Your approve is important for this lib

**Comment by AlexeiKislinskii10** [comment]
@prince-chrismc can Your clarify for me the process of review? I didn't see button for adding reviewers, so I can not add them by myself, and, if to be honest, I didn't know who can/should do that. And at least 3 reviewers should be for merging, but even after Your approve, even if uilianries will approve it(if he had access of course) there is not enough for merge. Can I do something to speedup this process?

**Comment by prince-chrismc** [comment]
checkout #2857 there's a few names you can tag from there... It also explains the rules

Unfortunately it's a best effort to go through all the PRs, this one has been on the slow side, but please be patient there's a lot of work open currently

File: recipes/resiprocate/all/conanfile.py
Comment (uilianries) [inline recipes/resiprocate/all/conanfile.py:34]:
```suggestion
            raise ConanInvalidConfiguration("reSIProcate is not support on {}."format(self.settings.os))
```

File: recipes/resiprocate/all/test_package/CMakeLists.txt
Comment (uilianries) [inline recipes/resiprocate/all/test_package/CMakeLists.txt:8]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

auto is C++11

**Review by uilianries — CHANGES_REQUESTED** [review]

**Comment by AlexeiKislinskii10** [comment]
@uilianries changes applied, thanks!

**Review by prince-chrismc — APPROVED** [review]

**Comment by AlexeiKislinskii10** [comment]
@madebr can Your be the last reviewer on this PR?
Summary of PR: adding new package with reSIProcate lib, version: 1.12.0, Windows and Mac are temporary unavailable due to build errors on this platforms. Works well on Linux(tested on Centos 8 and Debian 9)

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/resiprocate/all/conanfile.py
Comment (luismartingil) [inline recipes/resiprocate/all/conanfile.py:12]:
```suggestion
    url = "https://github.com/resiprocate/resiprocate"
```

File: recipes/resiprocate/all/conanfile.py
Comment (luismartingil) [inline recipes/resiprocate/all/conanfile.py:34]:
```suggestion
            raise ConanInvalidConfiguration("reSIProcate is not supported on {}.".format(self.settings.os))
```

File: recipes/resiprocate/all/test_package/test_package.cpp
Comment (luismartingil) [inline recipes/resiprocate/all/test_package/test_package.cpp:1]: @AlexeiKislinskii10 Is there a way you can add another test_package similar to https://github.com/uilianries/conan-resiprocate/tree/release/1.10.2/test_package ?

File: recipes/resiprocate/all/conanfile.py
Comment (madebr) [reply recipes/resiprocate/all/conanfile.py:12]: The url is correct.
Homepage is used for the package.

**Comment by prince-chrismc** [comment]
@uilianries you are blocking the PR from being merged, can you review when you get the change?

File: recipes/resiprocate/all/test_package/test_package.cpp
Comment (prince-chrismc) [reply recipes/resiprocate/all/test_package/test_package.cpp:1]: Thanks a noteworthy suggestion 👀 

The goal for CCI is to ensure the package is correct (headers/macros are present and links to libs) so we like to avoid the overly complicated testing library or example code... that's the projects responsibility.

Good thought 🥇  just not required here 😃

File: recipes/resiprocate/all/conanfile.py
Comment (AlexeiKislinskii10) [reply recipes/resiprocate/all/conanfile.py:12]: This field is managed by [this ](https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H027)rule, and is strict by content, we can not change it

File: recipes/resiprocate/all/test_package/test_package.cpp
Comment (AlexeiKislinskii10) [reply recipes/resiprocate/all/test_package/test_package.cpp:1]: Actually I can, with minimum of changes, just copy this test in new version of reSIProcate package, but is it needed? Test package should be simple to explain how to use this lib from conan.
@prince-chrismc is there any topic with rules about test package? Because I found only [this](https://docs.conan.io/en/latest/creating_packages/getting_started.html#the-test-package-folder) note

**Comment by AlexeiKislinskii10** [comment]
@uilianries sorry for double mention, but I have no choice, I really need this merge for my project. Hope Your are able to look at this PR again.

**Review by uilianries — APPROVED** [review]
LGTM. Good luck with your VoIP projects!

**Comment by AlexeiKislinskii10** [comment]
@prince-chrismc It is starange, I see 4 approvals, from you, uilianries, SSE4 and madebr, but automatic merge system detected only 2(ulianries and SSE4), so, as I see, not all approvments are valid for merging. I need to request someone new or I missed something?

**Comment by uilianries** [comment]
> @prince-chrismc It is starange, I see 4 approvals, from you, uilianries, SSE4 and madebr, but automatic merge system detected only 2(ulianries and SSE4), so, as I see, not all approvments are valid for merging. I need to request someone new or I missed something?

That's a bit weird, first I thought was the bot getting fuzzy, but actually I only see 2 approves. We need one more approve, however I also see 4 approves. Let me check it with Conan team. Sorry for this delay.

**Comment by prince-chrismc** [comment]
I think you might be refering to the github API
image


however the CCI bot uses (something close to)
image

it uses the REST api with a list from the community reviewers + CCI team to count them manually + apply custom logic

**Comment by jgsogo** [comment]
This is _waiting_ for a bugfix in the CI. @uilianries requested changes in a previous commit, then he approved the PR. The CI is not discarding the block with the new approval. We should deliver the patches this week, just a bit of patience, please 🙏 

Thanks!

**Pull request merged**
